File: SDKs/googleapis.com/recaptchaenterprise/v1/python/src/sdk/models/shared/googlecloudrecaptchaenterprisev1metrics.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import googlecloudrecaptchaenterprisev1challengemetrics as shared_googlecloudrecaptchaenterprisev1challengemetrics
from ..shared import googlecloudrecaptchaenterprisev1scoremetrics as shared_googlecloudrecaptchaenterprisev1scoremetrics


@dataclass_json
@dataclasses.dataclass
class GoogleCloudRecaptchaenterpriseV1Metrics:
    r"""GoogleCloudRecaptchaenterpriseV1Metrics
    Metrics for a single Key.
    """
    
    challenge_metrics: Optional[list[shared_googlecloudrecaptchaenterprisev1challengemetrics.GoogleCloudRecaptchaenterpriseV1ChallengeMetrics]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('challengeMetrics') }})
    name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('name') }})
    score_metrics: Optional[list[shared_googlecloudrecaptchaenterprisev1scoremetrics.GoogleCloudRecaptchaenterpriseV1ScoreMetrics]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('scoreMetrics') }})
    start_time: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('startTime') }})
    

File: SDKs/googleapis.com/recaptchaenterprise/v1/java/lib/src/main/java/openapisdk/models/shared/GoogleCloudRecaptchaenterpriseV1Metrics.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * GoogleCloudRecaptchaenterpriseV1Metrics
 * Metrics for a single Key.
**/
public class GoogleCloudRecaptchaenterpriseV1Metrics {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("challengeMetrics")
    public GoogleCloudRecaptchaenterpriseV1ChallengeMetrics[] challengeMetrics;
    public GoogleCloudRecaptchaenterpriseV1Metrics withChallengeMetrics(GoogleCloudRecaptchaenterpriseV1ChallengeMetrics[] challengeMetrics) {
        this.challengeMetrics = challengeMetrics;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("name")
    public String name;
    public GoogleCloudRecaptchaenterpriseV1Metrics withName(String name) {
        this.name = name;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("scoreMetrics")
    public GoogleCloudRecaptchaenterpriseV1ScoreMetrics[] scoreMetrics;
    public GoogleCloudRecaptchaenterpriseV1Metrics withScoreMetrics(GoogleCloudRecaptchaenterpriseV1ScoreMetrics[] scoreMetrics) {
        this.scoreMetrics = scoreMetrics;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("startTime")
    public String startTime;
    public GoogleCloudRecaptchaenterpriseV1Metrics withStartTime(String startTime) {
        this.startTime = startTime;
        return this;
    }
}